packages/era/src/e2s.ts

```diff
@@ -0,0 +1,184 @@
+
+  // Validate reserved bytes are zero (offset 6-7)
+  const reserved = header[6] | (header[7] << 8);
+  if (reserved !== 0) {
+    throw new Error(`E2Store reserved bytes must be zero, got: ${reserved}`);
+  }
+
+  return {type, length};
+}
+
+export async function readVersion(fh: FileHandle, offset: number): Promise<void> {
+  const versionHeader = new Uint8Array(E2STORE_HEADER_SIZE);
+  await fh.read(versionHeader, 0, E2STORE_HEADER_SIZE, offset);
+  if (Buffer.compare(versionHeader, VERSION_RECORD_BYTES) !== 0) {
+    throw new Error("Invalid E2Store version record");
+  }
+}
+
+/**
+ * Read a SlotIndex from a file handle.
+ */
+export async function readSlotIndex(fh: FileHandle, offset: number): Promise<SlotIndex> {
+  const recordEnd = offset;
+  const countBuffer = new Uint8Array(8);
+  await fh.read(countBuffer, 0, 8, recordEnd - 8);
+  const count = readInt48(countBuffer, 0);
+
+  const recordStart = recordEnd - (8 * count + 24);
+
+  // Validate index position is within file bounds
+  if (recordStart < 0) {
+    throw new Error(`SlotIndex position ${recordStart} is invalid - file too small for count=${count}`);
+  }
+
+  // Read and validate the slot index entry
+  const entry = await readEntry(fh, recordStart);
+  if (entry.type !== EntryType.SlotIndex) {
+    throw new Error(`Expected SlotIndex entry, got ${entry.type}`);
+  }
+
+  // Size: startSlot(8) + offsets(count*8) + count(8) = count*8 + 16
+  const expectedSize = 8 * count + 16;
+  if (entry.data.length !== expectedSize) {
+    throw new Error(`SlotIndex payload size must be exactly ${expectedSize} bytes, got ${entry.data.length}`);
+  }
+
+  // Parse start slot from payload
+  const startSlot = readInt48(entry.data, 0);
+
+  const offsets: number[] = [];
+  for (let i = 0; i < count; i++) {
+    offsets.push(readInt48(entry.data, 8 * i + 8));
+  }
+
+  return {
+    type: EntryType.SlotIndex,
+    startSlot,
+    offsets,
+    recordStart,
+  };
+}
+
+/**
+ * Write a single E2Store TLV entry (header + payload)
+ * Header layout: type[2] | length u32 LE | reserved u16(=0)
+ */
+export async function writeEntry(fh: FileHandle, offset: number, type: EntryType, payload: Uint8Array): Promise<void> {
+  const header = new Uint8Array(E2STORE_HEADER_SIZE);
+  // type
+  header[0] = type;
+  header[1] = type >>> 8;
+  // length u32 LE
+  header[2] = payload.length & 0xff;
+  header[3] = (payload.length >>> 8) & 0xff;
+  header[4] = (payload.length >>> 16) & 0xff;
+  header[5] = (payload.length >>> 24) & 0xff;
+  await fh.writev([header, payload], offset);
+}
+
+export async function writeVersion(fh: FileHandle, offset: number): Promise<void> {
+  await fh.write(VERSION_RECORD_BYTES, 0, VERSION_RECORD_BYTES.length, offset);
+}
+
+export function serializeSlotIndex(slotIndex: SlotIndex): Uint8Array {
+  const count = slotIndex.offsets.length;
+  const payload = new Uint8Array(count * 8 + 16);
+
+  // startSlot
+  writeInt48(payload, 0, slotIndex.startSlot);
```

Comment:
I didn't notice anywhere in the spec that this should be signed vs unsigned.  Do you have a reference you can point me to so that I can study up on this?